Knn_helper.py

```diff
@@ -0,0 +1,328 @@
+
+  # ins_ratio = ins_count / sum(_lib.crispr_subset(df)['Count'])
+  # alldf_dict['Ins1bp Ratio'].append(ins_ratio)
+
+  # seq, cutsite = _lib.get_sequence_cutsite(df)
+  # fivebase = seq[cutsite - 1]
+  # alldf_dict['Fivebase'].append(fivebase)
+
+  # _predict2.init_model()
+  # del_score = _predict2.total_deletion_score(seq, cutsite)
+  # alldf_dict['Del Score'].append(del_score)
+
+  # dlpred = _predict2.deletion_length_distribution(seq, cutsite)
+  # from scipy.stats import entropy
+  # norm_entropy = entropy(dlpred) / np.log(len(dlpred))
+  # alldf_dict['Entropy'].append(norm_entropy)
+
+  # local_seq = seq[cutsite - 4 : cutsite + 4]
+  # gc = (local_seq.count('C') + local_seq.count('G')) / len(local_seq)
+  # alldf_dict['GC'].append(gc)
+
+  # Store the Fivebase and threebase in df
+  # Both normally and one-hot encoded
+
+  # Get cutside by diving sequence length by 2
+  # TODO: Is this correct
+  cutsite = (int) (len(exp) / 2)
+
+  # Get fifth base and encode it
+  fivebase = exp[cutsite - 1]
+  alldf_dict['Fivebase'].append(fivebase)
+
+  if fivebase == 'A':
+    fivebase_oh = np.array([1, 0, 0, 0])
+  if fivebase == 'C':
+    fivebase_oh = np.array([0, 1, 0, 0])
+  if fivebase == 'G':
+    fivebase_oh = np.array([0, 0, 1, 0])
+  if fivebase == 'T':
+    fivebase_oh = np.array([0, 0, 0, 1])
+  alldf_dict['Fivebase_OH'].append(fivebase_oh)
+
+  # Get third base and encode it
+  threebase = exp[cutsite]
+  alldf_dict['Threebase'].append(threebase)
+  if threebase == 'A':
+    threebase_oh = np.array([1, 0, 0, 0])
+  if threebase == 'C':
+    threebase_oh = np.array([0, 1, 0, 0])
+  if threebase == 'G':
+    threebase_oh = np.array([0, 0, 1, 0])
+  if threebase == 'T':
+    threebase_oh = np.array([0, 0, 0, 1])
+  alldf_dict['Threebase_OH'].append(threebase_oh)
+
+  alldf_dict['_Experiment'].append(exp)
+
+  return alldf_dict
+
+def prepare_statistics(data_nm):
+  # Input: Dataset
+  # Output: Uniformly processed dataset, requiring minimal processing for plotting but ideally enabling multiple plots
+  # Calculate statistics associated with each experiment by name
+
+  alldf_dict = defaultdict(list)
+
+  alldf_dict_1bp = defaultdict(list)
+
+  timer = util.Timer(total = len(data_nm))
+
+  #TODO: Not sure if this is the idea?
+  insertion_data = data_nm[data_nm['Type'] == 'INSERTION'].reset_index()
+
+  # To make this run in a short time, take only the first n elements (i.e. [:n])
+  exps = insertion_data['Sample_Name'].unique()[:10]
+
+  for exp in exps:
+    ins_data = insertion_data[insertion_data['Sample_Name'] == exp]
+    print("seq: " + exp)
+    calc_statistics(ins_data, exp, alldf_dict)
+    calc_statistics_1bp(ins_data, exp, alldf_dict_1bp)
+    timer.update()
+
+  # Return a dataframe where columns are positions and rows are experiment names, values are frequencies
+  return pd.DataFrame(alldf_dict), pd.DataFrame(alldf_dict_1bp)
+
+##
+# Run statistics
+##
+def calc_statistics_1bp(df, exp, alldf_dict):
```

Comment:
I would delete all the commented parts for now, I think we need this only once the knn is trained
```suggestion
def calc_statistics_1bp(df, exp, alldf_dict):
  pass
```